File: src/main/java/com/example/springwsdemo/NumberController.java
package com.example.springwsdemo;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.math.BigDecimal;

@RestController
public class NumberController {

    private final NumberConversionClient numberConversionClient;

    public NumberController(NumberConversionClient numberConversionClient) {
        this.numberConversionClient = numberConversionClient;
    }

    @GetMapping("/number/{input}")
    String convert(@PathVariable String input) {
        return numberConversionClient.convert(new BigDecimal(input));
    }
}

File: src/main/kotlin/com/example/springwsdemo/NumberController.kt
package com.example.springwsdemo

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import java.math.BigDecimal

@RestController
class NumberController(private val numberConversionClient: NumberConversionClient) {
    @GetMapping("/tax/{input}/{taxRate}")
    fun convert(@PathVariable input: String, @PathVariable taxRate: String): String {
        val inputNumber = BigDecimal(input).setScale(2)
        val taxRateNumber = BigDecimal(taxRate).setScale(2)
        val inputWords = numberConversionClient.convert(inputNumber)
        val wordsWithTax = numberConversionClient.convert(
            inputNumber + inputNumber * taxRateNumber / BigDecimal(100))
        return "$inputWords plus $taxRate% tax is $wordsWithTax"
    }
}